src/createReduxForm.js

```diff
@@ -670,13 +674,19 @@ const createReduxForm = (structure: Structure<*, *>) => {
               ? values
               : setIn(values, name, value)
             const syncValidationPasses = submitting || !getIn(syncErrors, name)
+            const fieldNeedsValidationForBlur =
+              asyncBlurFields &&
+              ~asyncBlurFields.indexOf(name.replace(/\[[0-9]+\]/g, '[]'))
+            const fieldNeedsValidationForChange =
+              asyncChangeFields &&
+              ~asyncChangeFields.indexOf(name.replace(/\[[0-9]+\]/g, '[]'))
```

Comment:
@shepmaster [pointed out](https://github.com/erikras/redux-form/pull/3531/files/270ea7b1114ff57cbb74e3bdf0faac4ccc575b3f#r154512179) on a PR of mine that `asyncValidate` gets called in one location without any arguments, causing `name.replace` to throw an error. Although this code has now been modified, I expect this to still be an issue on lines 679 & 682.